keras/src/export/litert.py

```diff
@@ -0,0 +1,255 @@
+            filepath: Output path for the exported model
+
+        Returns:
+            Path to exported model
+        """
+        # 1. Resolve / infer input signature
+        if self.input_signature is None:
+            # Try dict-specific inference first (for models with dict inputs)
+            dict_signature = self._infer_dict_input_signature()
+            if dict_signature is not None:
+                self.input_signature = dict_signature
+            else:
+                # Fall back to standard inference
+                self.input_signature = get_input_signature(self.model)
+
+        # 3. Handle dictionary inputs by creating an adapter
+        # Check if we have dict inputs that need adaptation
+        has_dict_inputs = isinstance(self.input_signature, dict)
+
+        if has_dict_inputs:
+            # Create adapter model that converts list to dict
+            adapted_model = self._create_dict_adapter(self.input_signature)
+
+            # Convert dict signature to list for TFLite conversion
+            # The adapter will handle the dict->list conversion
+            input_signature_list = list(self.input_signature.values())
+
+            # Use adapted model and list signature for conversion
+            model_to_convert = adapted_model
+            signature_for_conversion = input_signature_list
+        else:
+            # No dict inputs - use model as-is
+            model_to_convert = self.model
+            signature_for_conversion = self.input_signature
+
+        # Store original model reference for later use
+        original_model = self.model
+
+        # Temporarily replace self.model with the model to convert
+        self.model = model_to_convert
+
+        try:
+            # 4. Convert the model to TFLite.
+            tflite_model = self._convert_to_tflite(signature_for_conversion)
+        finally:
+            # Restore original model
+            self.model = original_model
+
+        # 4. Save the initial TFLite model to the specified file path.
+        if not filepath.endswith(".tflite"):
+            raise ValueError(
+                "The LiteRT export requires the filepath to end with "
+                "'.tflite'. Got: {filepath}"
+            )
+
+        with open(filepath, "wb") as f:
+            f.write(tflite_model)
+
+        return filepath
+
+    def _create_dict_adapter(self, input_signature_dict):
+        """Create an adapter model that converts list inputs to dict inputs.
+
+        This adapter allows models expecting dictionary inputs to be exported
+        to TFLite format (which only supports positional/list inputs).
+
+        Args:
+            input_signature_dict: Dictionary mapping input names to InputSpec
+
+        Returns:
+            A Functional model that accepts list inputs and converts to dict
+        """
+        io_utils.print_msg(
+            f"Creating adapter for dictionary inputs: "
+            f"{list(input_signature_dict.keys())}"
+        )
```

Comment:
Instead, print the mapping from the nested structure to the flat list so that people know in which order to put their inputs in the list.

```python
paths_and_specs = keras.tree.flatten_with_path(input_signature_struct)
paths = [".".join(str(e) for e in p) for v, p in paths_and_specs]
io_utils.print_msg(f"Creating adapter for inputs: {paths}")
```